pkg/cache/snapshot_test.go

```diff
@@ -40,149 +39,208 @@ func TestSnapshot(t *testing.T) {
 	if err := kueue.AddToScheme(scheme); err != nil {
 		t.Fatalf("Failed adding kueue scheme: %s", err)
 	}
-	cache := New(fake.NewClientBuilder().WithScheme(scheme).Build())
-	clusterQueues := []kueue.ClusterQueue{
+
+	cases := []struct {
```

Comment:
```suggestion
	cases := map[string]struct {
```

This way, the cases might run in a different order each time, which ensure that they don't depend on each other.